### PR TITLE
General catch all oh shit#!"% error handler 

### PR DESCRIPTION
**Pull request by johanherman**
Even for a general catch all error handler there seems to be some weird "behaviour" (gah!) when using multiple nested workflows. The interface wouldn't show on what step in the subflow that failed regardless how I did it, so for now I've decided to let everything to be in one big workflow instead of having several subflows...

### CODE REVIEW

File: actions/oh_shit_error.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    description: "Host where the runfolder is located."
+
+  runfolder:
+    default: null
+    type: string
+    required: true
+    description: "Name of runfolder that failed"
+
+  send_mail_to:
+    default: johan.hermansson@medsci.uu.se
```
Comment (johandahlberg) [inline actions/oh_shit_error.yaml]: Should we change the default address here (and in other places) to examples so you and our list don't have to be spammed if some other user picks this up?

**Comment by johandahlberg** [comment]
I think that if we put everything in the same workflow now maybe we should remove the rest of the files? For now at least I think that keeping it all in the same file will be manageable as there aren't to many steps?  Also we might want to do a general clean up?

File: actions/oh_shit_error.yaml (hunk 1 above)
Comment (johanherman) [inline actions/oh_shit_error.yaml]: That we can do.

**Comment by johanherman** [comment]
Yeah, there's a bunch of unused things in there. I'll clean that up..

**Comment by johanherman** [comment]
Removed some stuff now, and have set `remote_host` and `send_mail_to` to some non-valid values that have to be changed. 

I guess we have to re-make our deployment of stackstorm-master though. At the moment we're mounting everything within the guest host with Vagrant, which means that we can't simply regexp out the values from the system-admin orchestration.

**Comment by johandahlberg** [comment]
I see your point. What do you by the way think about putting all of those things into the config file, and then using a python action to inject it into the context? I favor setting anything that is relatively static (e.g. ports) that way and just setting things which are dynamic (and changed often) in the workflow yaml files.

**Comment by johandahlberg** [comment]
Also this is currently at -425. That's nice. :)

File: actions/workflows/ngi_uu_workflow.yaml [hunk 2]
```diff
@@ -12,96 +11,157 @@ workflows:
...
+            - barcode_mismatches
+            - tiles
+            - use_base_mask
+            - bcl2fastq_additional_arguments
+            - siswrap_service_port
         output:
             output_the_whole_workflow_context: <% $ %>
+        task-defaults:
+            on-error:
+                - oh_shit_error
```
Comment (johandahlberg) [inline actions/workflows/ngi_uu_workflow.yaml]: I feel your pain...

**Comment by johandahlberg** [comment]
Great job with this!

**Pull request merged**